subsys/net/l2/ethernet/gptp/gptp_data_set.h

```diff
@@ -376,6 +377,26 @@ struct gptp_time_prop_ds {
 	bool freq_traceable : 1;
 };
 
+/**
+ * @brief Servo Controllor Data Set.
+ *
+ * Data Set representing time sync servo controllor.
+ */
+struct gptp_servo_ds {
```

Comment:
Disclaimer: I'm not a PTP (or gPTP) expert, so please let me know if I misunderstand concepts from the standards which I'm still learning myself currently.

That being said, I believe that the term "data set" has a well defined meaning in the context of IEEE 1588 and IEEE 802.1AS-2011. Therefore I propose one of two changes:
* Either you should point to standard clauses that justify your addition in this file or
* you should neither call non-standard data objects "data sets" nor should they be added to a file that implements standard objects for Zephyr.

This is relevant as the standardized data sets are used in higher-level building blocks, e.g. standardized management protocols, profiling, etc.

It seems that others have deviated from the standard in this file as well, so either I'm wrong or that's to be considered technical debt as well (not to be fixed in this PR of course). @jukkar What's your opinion?

Also it seems you let yourself be inspired by linuxptp (which is great). As you can see there, servos are an exchangeable concept as there is no "one best solution" for clock syntonization/synchronization (see PTP standard for definitions).

Of course we don't need an over-engineered multi-servo/multi-profile solution while we only have one algorithm but we should at least be prepared for the following use cases in the sense of not building up unnecessary "barriers to entry" to future contributors:
* multiple alternative syntonization/synchronization mechanisms (also see my clock RFC for re-use in network/POSIX/libc system clocks and compare Linux's adjtimex/clock_adjtime approach which linuxptp connects to). 
* extending code with additional features specified in the standards - namely in this context data set management (IEEE 1588-2019, sections 8.1.4 and 15, IEEE 802.1AS-2011 sections 14 and 15, RFC 1873).
* remain fully compatible with PTP (IEEE 1588) in the sense of IEEE 802.1AS being just one of many possible PTP profiles. For standard compliance, syntonization/synchronization algorithms are configurable properties of PTP profiles AFAICS.

This all should not at all result in huge changes nor should it introduce any additional complexity. We just need to ensure to keep the door open (think global, act local). I'm of course willing and available to invest considerable "thinking time" myself such that we find a really lean but fully extensible solution.

Just my 2c - please let me know if you think I'm wrong. I need to run, but I'd love to do a more detailed review later today. Hope the PR can wait for a day - let me know if it is somehow urgent.

Reply:
> I believe that the term "data set" has a well defined meaning in the context of IEEE 1588 and IEEE 802.1AS-2011: You should neither call non-standard data objects "data sets" nor should they be added to a file that implements standard objects for Zephyr. I propose you put them in servo_pi.c/h files and create a properly prefixed gptp_servo_pi_state struct for internal state. See my other comments re prefixing.
> 
> This is relevant as the standardized data sets are used in higher-level building blocks, e.g. standardized management protocols, profiling, etc. It seems that others have deviated from the standard in this file as well (mixing up data sets and global sm variables, @jukkar do you agree?), but I'd consider that technical debt not to be continued but out-of-scope for this PR.

Yes, I agree with you here. Better not call non standard data objects with same name as standard ones.